engine/src/main/java/org/pentaho/di/kitchen/Kitchen.java

```diff
@@ -221,6 +220,23 @@ public KettleException call() throws Exception {
       throw repositoryRegisterException;
     }
     Future<KettleException> kettleInitFuture = repositoryRegisterResults.getValue();
+    List<CommandLineOption> updatedOptionList = getAdditionalCommandlineOption( options, pluginParam );
+    options = updatedOptionList.toArray( new CommandLineOption[ 0 ] );
+    LogChannelInterface log = new LogChannel( STRING_KITCHEN );
+
+    if ( args.size() == 2 ) {
+      CommandLineOption.printUsage( updatedOptionList.toArray( new CommandLineOption[ 0 ] ) );
+      blockAndThrow( kettleInitFuture );
+      exitJVM( 9 );
+    }
+
+    CommandLineOption.parseArguments( args, options, log );
+
+    if ( !Utils.isEmpty( optionRepname.toString() ) && !Utils.isEmpty( pluginParam.toString() ) ) {
+      log.logError( BaseMessages.getString( PKG, "Kitchen.Error.ComdLine.RepAndProject" ) );
+      blockAndThrow( kettleInitFuture );
+      exitJVM( 9 );
+    }
```

Comment:
If we can't do this in the plugin, we should just skip the check. 

It seems like repository+project is going to be required shortly anyway.